.github/workflows/deno.yml

```diff
@@ -7,25 +7,28 @@ on:
     branches: [main]
 
 jobs:
-  fmt-lint-check-test:
+  denofmt-and-lint:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
       - uses: denoland/setup-deno@v1
         with:
           deno-version: v1.x
+      - uses: actions/checkout@v3
 
-      - name: Format
-        run: deno fmt --check
-
-      - name: Lint
-        run: deno lint
+      - run: deno lint
+      - run: deno fmt --check
 
-      - name: Check
-        run: deno task check
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: denoland/setup-deno@v1
+        with:
+          deno-version: v1.x
+      - uses: actions/checkout@v3
 
-      - name: Test
-        run: deno test
+      - run: deno cache source/*.ts
+      - run: deno task check
+      - run: deno test
```

Comment:
Redundant.

```suggestion
      - run: deno test
```

Reply:
the benefit of splitting it up is easier visibility of errors on the CI interface. when the cache fails its seen directly. Also the time taken for each of the steps is easier distinguishable.

deno test doesn't do the checks anymore when not specified with --check stuff? also check checks not only the tests?

Reply:
It checks tests and local files imported by them: https://deno.com/manual/getting_started/command_line_interface#type-checking-flags

Reply:
But this would require to have every file included in some test?

Reply:
Tests already import `mod.ts`.

Reply:
in this case yeah :thinking: But in general I wouldnt be so sure about my testing efforts there :D